### ISSUE TITLE
YOLOV8 train all loss is NAN and P/R/map is 0 whether GPU or CPU

### ISSUE THREAD

**Issue by Cat2eacher** [ultralytics/ultralytics#14757]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

在训练过程中，无论是CPU还是GPU，Loss都是NAN，P,R,map值都是0。
尝试了修改amp，修改half，仍然无法解决问题。
想分享记录一下自己的解决办法。
During the training, both CPU and GPU, Loss is Nan, p, R, and map values are 0.
Tried modifying amp, modified half, still couldn't fix the problem.
Want to share and document your solutions.

### Environment

GPU is GTX 2080. CPU is AMD R7 6800H, CUDA 11.4
When I follow the official document, 【pip install ultralytics】
The default torch is the latest version 2.4.0！

### Minimal Reproducible Example

pip uninstall torch
pip uninstall torchvision

# CUDA 11.3
pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 torchaudio==0.12.1 --extra-index-url https://download.pytorch.org/whl/cu113
# CPU only
pip install torch==1.12.1+cpu torchvision==0.13.1+cpu torchaudio==0.12.1
# 重装Pytorch后就可以啦！
# Just reinstall the PYTORCH!

### Additional

Maybe it's the torch

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by pderrenger** [comment]
@Cat2eacher thank you for sharing your experience. The issue you're encountering with NaN losses and zero P/R/mAP values during training is often related to data or environment setup. Please ensure your dataset is correctly formatted and that there are no anomalies in your labels. Additionally, verify that your training configuration is appropriate for your dataset.

Given that you suspect the PyTorch version might be the cause, I recommend trying the latest versions of both Ultralytics and PyTorch to see if the issue persists. You can update Ultralytics with `pip install --upgrade ultralytics` and PyTorch following the official installation guide for your CUDA version.

If the problem continues, please provide more details about your dataset and training configuration. This will help us diagnose the issue more effectively.

**Comment by Y-T-G** [comment]
Try reducing `lr0`.

Check for any warnings in the training logs.

**Comment by AliBYetiser** [comment]
is your data properly annotated?

**Comment by glenn-jocher** [comment]
@AliBYetiser please ensure your dataset is correctly annotated and follows the expected format. Incorrect or poor-quality annotations can lead to issues like NaN losses and zero metrics.

**Comment by glenn-jocher** [comment]
Please ensure your dataset is correctly annotated and follows the expected format. Incorrect or poor-quality annotations can lead to issues like NaN losses and zero metrics.
